meson.build

```diff
@@ -0,0 +1,176 @@
+####################################################################################################################################
+# pgBackRest Project
+####################################################################################################################################
+project(
+    'pgbackrest',
+    ['c'],
+    version: '2.40dev',
+    license: 'MIT',
+    meson_version: '>=0.53.2',
+    default_options: [
+        # Core options
+        'buildtype=release',
+        'warning_level=2',
+
+        # Base options
+        'b_ndebug=if-release',
+
+        # Compiler options
+        'c_std=c99',
+    ],
+)
+
+####################################################################################################################################
+# OS-specific settings
+####################################################################################################################################
+if host_machine.system() == 'linux'
+    add_global_arguments('-D_POSIX_C_SOURCE=200809L', language : 'c')
+elif host_machine.system() == 'darwin'
+    add_global_arguments('-D_DARWIN_C_SOURCE', language : 'c')
+endif
+
+####################################################################################################################################
+# Enable as many additional warnings as possible to catch potential errors
+####################################################################################################################################
+warning_enable = [
```

Comment:
Hello, 
I think that this array, (or maybe just the `68: foreach warning : warning_enable` below, needs to be checked against the OS, as this opens way to add similar warnings to other compilers, mainly msvc. 

Aside from that, I think it is great.

Reply:
I'd prefer to hold off on that for now. We'll add that in your branch once we know what the Windows builds are going to look like. If we add it now it won't have a purpose and will probably end up looking different.

Reply:
For example, I don't think we want to do an OS check here, but a compiler check. But we'll figure that out as we work on the Windows port.

Reply:
You already use compiler.has_argument() here, which uses a configure probe to check whether the compiler supports it. So you can add both GCC and MSVC style arguments here and it will filter out whichever ones don't work.

It's possible to check the OS / compiler manually via host_machine.system() or compiler.get_id(), but all that will do is make the configure stage a bit faster as it skips compiling a test file to see if the command line argument triggered a fatal "invalid argument" error, on compilers that you statically code as "doesn't support this particular argument".

Reply:
Excellent suggestion -- and exactly why I want to see how the Windows build works out before make changes to support it.